icu4c/source/i18n/numfmt.cpp

```diff
@@ -1310,6 +1310,14 @@ NumberFormat::makeInstance(const Locale& desiredLocale,
         status = U_ILLEGAL_ARGUMENT_ERROR;
         return NULL;
     }
+    
+    // For the purposes of general number formatting, UNUM_NUMBERING_SYSTEM should behave the same
+    // was as UNUM_DECIMAL.  In both cases, you get either a DecimalFormat or a RuleBasedNumberFormat
+    // depending on the locale's numbering system (either the default one for the locale or a specific
+    // one specified by using the "@numbers=" or "-u-nu-" parameter in the locale ID.
+    if (style == UNUM_NUMBERING_SYSTEM) {
+        style = UNUM_DECIMAL;
+    }
```

Comment:
As suggested below: I feel like this is a bug in the RBNF constructor, not in the `unum_open` or `NumberFormat::makeInstance` wrappers. I'm a little worried about changing around the old behavior.